share/functions/__fish_cached.fish

```diff
@@ -0,0 +1,40 @@
+function __fish_cached --description "Cache the command output for a given amount of time"
+
+    argparse --min-args 1 --stop-nonopt 't/max-age=!_validate_int --min 0' 'k/cache-key=' -- $argv
+    or return
+
+    if set -q _flag_cache_key
+        set -f cache_key $_flag_cache_key
+    else
+        set -f cache_key (string trim "$argv" | string split ' ')[1]
+    end
+    if not string match -q --regex '^[\w+-]+$' -- "$cache_key"
+        return 1
+    end
+
+    if set -q _flag_max_age
+        set -f max_age $_flag_max_age
+    else
+        set -f max_age -1
+    end
+    set -l cache_dir (__fish_make_cache_dir)
+    or return
+    set -l cache_file (path normalize $cache_dir/$cache_key)
+    set -l cache_age (path mtime --relative $cache_file)
+
+    if not test -f $cache_file
+        __fish_cache_put $cache_file
+        eval "$argv >$cache_file &"
```

Comment:
Possibly we could rework this so that these 2 lines don't need to be repeated on lines 36 and 37, but the current form seems to make easier to understand what is going on here.